ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -362,21 +365,26 @@ protected Node visitJoin(final Join node, final AnalysisContext context) {
 
   private StructuredDataSource timestampColumn(
       AliasedRelation aliasedRelation,
-      StructuredDataSource structuredDataSource
-  ) {
-    if (((Table) aliasedRelation.getRelation()).getProperties() != null) {
-      if (((Table) aliasedRelation.getRelation()).getProperties()
-              .get(DdlConfig.TIMESTAMP_NAME_PROPERTY) != null) {
-        String timestampFieldName = (((Table) aliasedRelation.getRelation()))
-            .getProperties().get(DdlConfig.TIMESTAMP_NAME_PROPERTY).toString().toUpperCase();
-        if (!(timestampFieldName.startsWith("'") && timestampFieldName.endsWith("'"))) {
-          throw new KsqlException("Property name should be String with single qoute.");
-        }
-        timestampFieldName = timestampFieldName.substring(1, timestampFieldName.length() - 1);
-        structuredDataSource = structuredDataSource.cloneWithTimeField(timestampFieldName);
-      }
-    }
-    return structuredDataSource;
+      StructuredDataSource structuredDataSource) {
+    final Map<String, Expression> properties
+        = ((Table) aliasedRelation.getRelation()).getProperties();
+
+    if (properties == null) {
+      return structuredDataSource;
+    }
+
+    final String timestampName = properties.containsKey(DdlConfig.TIMESTAMP_NAME_PROPERTY)
+        ? properties.get(DdlConfig.TIMESTAMP_NAME_PROPERTY).toString()
```

Comment:
Should we remove the quotes?

Reply:
They get removed in `TimestampExtractionPolicyFactory.create(...)`